File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/Tabs/StatTab/Model/AnalyseCardViewHolder.java
package com.teamunemployment.lolanalytics.front_page.Tabs.StatTab.Model;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.github.mikephil.charting.charts.PieChart;
import com.github.mikephil.charting.components.Description;
import com.github.mikephil.charting.components.Legend;
import com.github.mikephil.charting.data.PieData;
import com.github.mikephil.charting.data.PieDataSet;
import com.github.mikephil.charting.data.PieEntry;
import com.teamunemployment.lolanalytics.front_page.Tabs.StatTab.AnalyseTabContract;
import com.teamunemployment.lolanalytics.R;

import java.util.ArrayList;
import java.util.List;


import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

import static com.github.mikephil.charting.utils.ColorTemplate.rgb;

/**
 * Created by Josiah Kendall
 *
 * This is the view item for
 */

public class AnalyseCardViewHolder extends RecyclerView.ViewHolder implements AnalyseTabContract.CardView {

    private int position;
    private AnalyseTabContract.Presenter presenter;

    @BindView(R.id.performance_pie) PieChart pieChart;
    @BindView(R.id.analyse_card_title) TextView titleTextView;
    @BindView(R.id.performance_change) TextView performanceChange;

    public AnalyseCardViewHolder(View itemView, AnalyseTabContract.Presenter analysePresenter) {
        super(itemView);
        ButterKnife.bind(this ,itemView);
        this.presenter = analysePresenter;
    }

    @Override
    public void SetTitle(String title) {
        titleTextView.setText(title);
    }

    @Override
    public void SetGraph(float enemyStat, float heroStat) {
        // set the performance pie here
        List<PieEntry> entries = new ArrayList<>();

        entries.add(new PieEntry(enemyStat, "Opponent performance"));
        entries.add(new PieEntry(heroStat, "My performance"));


        PieDataSet set = new PieDataSet(entries, "");
        int[] MATERIAL_COLORS = {
                rgb("#FF3D00"),rgb("#00E676")
        };
        set.setColors(MATERIAL_COLORS);
        set.setDrawValues(false);
        PieData data = new PieData(set);
        pieChart.setData(data);
        pieChart.setDrawMarkers(false);
        pieChart.setUsePercentValues(false);
        pieChart.setDrawSliceText(false);
        pieChart.setHighlightPerTapEnabled(false);
        Float floater = heroStat;

        pieChart.setCenterText(Integer.toString(floater.intValue()).concat("%"));
        pieChart.setCenterTextColor(R.color.colorAccent);
        pieChart.setDrawEntryLabels(false);

        // Dont want to display a description.
        Description description = new Description();
        description.setText("");
        pieChart.setDescription(description);

        // Customize legend.
        pieChart.getLegend().setForm(Legend.LegendForm.CIRCLE);
        pieChart.getLegend().setEnabled(false);
        pieChart.invalidate();

    }

    @Override
    public void SetChange(double change) {
        performanceChange.setText(Double.toString(change));
    }

    @Override
    public void SetItemPosition(int position) {
        this.position = position;
    }

    @OnClick(R.id.root_analyse_item)
    public void handleClick() {
        presenter.HandleItemClick(position);
    }
}

File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/Tabs/StatTab/Model/AnalyseCardViewHolder.kt
package com.teamunemployment.lolanalytics.front_page.Tabs.StatTab.Model

import android.support.v7.widget.RecyclerView
import android.view.View

import com.github.mikephil.charting.components.Description
import com.github.mikephil.charting.components.Legend
import com.github.mikephil.charting.data.PieData
import com.github.mikephil.charting.data.PieDataSet
import com.github.mikephil.charting.data.PieEntry
import com.teamunemployment.lolanalytics.front_page.Tabs.StatTab.AnalyseTabContract
import com.teamunemployment.lolanalytics.R

import java.util.ArrayList

import com.github.mikephil.charting.utils.ColorTemplate.rgb
import kotlinx.android.synthetic.main.analyse_card_item.view.*

/**
 * Created by Josiah Kendall
 *
 * This is the view item for
 */

class AnalyseCardViewHolder(private val view: View, private val presenter: AnalyseTabContract.Presenter) : RecyclerView.ViewHolder(view), AnalyseTabContract.CardView {

    var itemPosition: Int = 0

    init {
        view.cardBase.setOnClickListener { handleCardClick() }
    }

    private val pieChart = itemView.earlyGame

    override fun setTitle(title: String) {
        view.analyseCardTitle.text = title
    }

    override fun setEarlyGame(hero: Float, enemy: Float) {

    }

    override fun setMidGame(hero: Float, enemy: Float) {

    }

    override fun setLateGame(hero: Float, enemy: Float) {

    }

    //    @Override
    @Deprecated("")
    fun SetGraph(enemyStat: Float, heroStat: Float) {

        // set the performance pie here
        val entries = ArrayList<PieEntry>()

        entries.add(PieEntry(enemyStat, "Opponent performance"))
        entries.add(PieEntry(heroStat, "My performance"))

        val set = PieDataSet(entries, "")
        val MATERIAL_COLORS = intArrayOf(rgb("#FF3D00"), rgb("#00E676"))
        set.setColors(*MATERIAL_COLORS)
        set.setDrawValues(false)
        val data = PieData(set)
        pieChart.data = data
        pieChart.setDrawMarkers(false)
        pieChart.setUsePercentValues(false)
        pieChart.setDrawSliceText(false)
        pieChart.isHighlightPerTapEnabled = false

        pieChart.centerText = Integer.toString(heroStat.toInt()) + "%"
        pieChart.setCenterTextColor(R.color.colorAccent)
        pieChart.setDrawEntryLabels(false)

        // Dont want to display a description.
        val description = Description()
        description.text = ""
        pieChart.description = description

        // Customize legend.
        pieChart.legend.form = Legend.LegendForm.CIRCLE
        pieChart.legend.isEnabled = false
        pieChart.invalidate()
    }

    //    @Override
    @Deprecated("")
    fun setChange(change: Double) {
//        performanceChange!!.text = java.lang.Double.toString(change)
    }

    private fun handleCardClick() {
        presenter.handleItemClick(position)
    }
}
